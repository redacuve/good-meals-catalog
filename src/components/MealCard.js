import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function MealCard(props) {
  const { color, meal } = props;
  return (
    <Link
      to={`/meal/${meal.idMeal}`}
      key={meal.strMeal}
      className="w-full md:w-1/2 lg:w-1/3 xl:w-1/4 p-1 mb-2 shadow border rounded bg-white mb-5 cursor-default"
    >
      <div className="w-full">
        <img className="w-full" src={meal.strMealThumb} alt={meal.strMeal} />
      </div>
      <div className={`w-full bg-${color}-300`}>
        <h3 className="font-bold text-2xl pl-3">{meal.strMeal}</h3>
      </div>
      <div className="w-full flex justify-center mt-3">
        <div className={`bg-transparent hover:bg-${color}-500 text-${color}-700 text-sm font-semibold hover:text-white py-2 px-4 border border-${color}-500 hover:border-transparent rounded cursor-pointer`}>
          View Meal
        </div>
      </div>
    </Link>
  );
}

MealCard.propTypes = {
  color: PropTypes.string.isRequired,
  meal: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default MealCard;
