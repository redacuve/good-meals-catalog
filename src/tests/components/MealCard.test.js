import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import MealCard from '../../components/MealCard';

afterEach(cleanup);

const color = 'indigo';
const meal = {
  strMeal: 'Crock Pot Chicken Baked Tacos',
  strMealThumb:
    'https://www.themealdb.com/images/media/meals/ypxvwv1505333929.jpg',
  idMeal: '52830',
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <BrowserRouter>
      <MealCard color={color} meal={meal} />
    </BrowserRouter>,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('render MealCard icon correctly', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <MealCard color={color} meal={meal} />
    </BrowserRouter>,
  );
  expect(getByTestId('mealcard-title')).toContainHTML(
    'Crock Pot Chicken Baked Tacos',
  );
});

it('render container correctly', () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <MealCard color={color} meal={meal} />
    </BrowserRouter>,
  );
  const child = getByTestId('mealcard-link');
  expect(getByTestId('mealcard')).toContainElement(child);
});

it('matches snapshot 1', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <MealCard color={color} meal={meal} />
      </BrowserRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
