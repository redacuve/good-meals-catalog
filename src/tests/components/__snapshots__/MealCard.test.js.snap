// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<a
  className="w-full md:w-1/2 lg:w-1/3 xl:w-1/4 p-1 mb-2 shadow border rounded bg-white mb-5 cursor-default"
  data-testid="mealcard"
  href="/meal/52830"
  onClick={[Function]}
>
  <div
    className="w-full"
  >
    <img
      alt="Crock Pot Chicken Baked Tacos"
      className="w-full"
      src="https://www.themealdb.com/images/media/meals/ypxvwv1505333929.jpg"
    />
  </div>
  <div
    className="w-full bg-indigo-300"
  >
    <h3
      className="font-bold text-2xl pl-3"
      data-testid="mealcard-title"
    >
      Crock Pot Chicken Baked Tacos
    </h3>
  </div>
  <div
    className="w-full flex justify-center mt-3"
    data-testid="mealcard-link"
  >
    <div
      className="bg-transparent hover:bg-indigo-500 text-indigo-700 text-sm font-semibold hover:text-white py-2 px-4 border border-indigo-500 hover:border-transparent rounded cursor-pointer"
    >
      View Meal
    </div>
  </div>
</a>
`;
